### PR TITLE
Spring Boot Dependency Upgrade

### PR DESCRIPTION
**Pull request by cachescrubber**
## Spring Boot Dependency Upgrade

Upgrade Spring Generator Dependencies

- Upgrade to Spring Boot 2.6.5 / 2.5.11
- Upgrade to Springdoc 1.6.6 
- Upgrade to Swagger Annotations 1.6.5 / 2.1.13
- Upgrade to Swagger Ui 4.8.1

Some minor cosmetic changes to the spring-boot pom in order to use the generated project as a dependency:

- Set relativePath in parent section
- Support a classifier in the spring-boot maven plugin configuration

### PR checklist
 
- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by cachescrubber** [comment]
@welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

FYI

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache [hunk 1]
```diff
@@ -40,7 +40,8 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>{{#springFoxDocumentationProvider}}2.5.8{{/springFoxDocumentationProvider}}{{^springFoxDocumentationProvider}}2.6.3{{/springFoxDocumentationProvider}}</version>
+        <version>{{#springFoxDocumentationProvider}}2.5.10{{/springFoxDocumentationProvider}}{{^springFoxDocumentationProvider}}2.6.4{{/springFoxDocumentationProvider}}</version>
+        <relativePath/> <!-- lookup parent from repository -->
```
Comment (welshm) [inline modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache:44]: Is this required or just recommended to add?

Comment (cachescrubber) [reply modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache:44]: It's the way [start.spring.io](https://start.spring.io) recommends. It is required when you have a multi-module maven project where the generated model is a sub module.

**Comment by cachescrubber** [comment]
re-open to trigger ci

**Comment by wing328** [comment]
LGTM

**Pull request merged**